diff -Naur freecad-0.14.3702.orig/src/Gui/CMakeLists.txt freecad-0.14.3702/src/Gui/CMakeLists.txt
--- freecad-0.14.3702.orig/src/Gui/CMakeLists.txt	2014-07-13 10:33:02.000000000 -0500
+++ freecad-0.14.3702/src/Gui/CMakeLists.txt	2014-07-22 21:25:04.508664979 -0500
@@ -57,6 +57,7 @@
     set(FreeCADGui_LIBS
         ${FreeCADGui_LIBS}
         ${SPNAV_LIBRARIES}
+        X11
     )
 ENDIF(SPNAV_FOUND)
 
